tests/test_intervention_outcome.py

```diff
@@ -0,0 +1,160 @@
+from conftest import *
+from datetime import datetime
+
+from models.appendix import InterventionReason
+from models.prescription import Intervention
+from security.role import Role
+
+import json
+from collections.abc import Mapping
+
+# import auxiliary dictionaries to this test
+from tests.aux_test_intervention import *
+
+
+def get_nested_value(data, path):
+    keys = path.split(".")
+    for key in keys:
+        if isinstance(data, Mapping) and key in data:
+            data = data[key]
+        elif isinstance(data, list) and key.isdigit():
+            data = data[int(key)]
+        else:
+            return None
+    return data
+
+
+def compare_specific_fields(data1, data2, fields_to_compare):
+    differences = {}
+
+    for field in fields_to_compare:
+        value1 = get_nested_value(data1, field)
+        value2 = get_nested_value(data2, field)
+
+        if value1 is None and value2 is None:
+            continue
+        elif value1 is None:
+            differences[field] = ("Field not present in first JSON", value2)
+        elif value2 is None:
+            differences[field] = (value1, "Field not present in second JSON")
+        elif value1 != value2:
+            differences[field] = (value1, value2)
+
+    return differences
+
+
+def compare_json_fields(json_str1, json_str2, fields_to_compare):
+
+    return compare_specific_fields(json_str1, json_str2, fields_to_compare)
+
+
+# Criar nova intervenção
+@pytest.fixture
+def test_put_interventions(client):
+    """Teste put /intervention  e retorna idIntervention"""
+    access_token = get_access(client, roles=[Role.PRESCRIPTION_ANALYST.value])
+    idPrescription = "198"

```

Comment:
acho desnecessário criar as variáveis para depois mencionar elas apenas na declaração do dict data.

Reply:
I did this way, to make it simpler to extend the test with different inputs using  @pytest.mark.parametrize.
But, followed your recommendation.